### PR TITLE
#542: Add vagrant setup for libvirt

### LINKED ISSUE #542: Add Vagrant Setup for libvirt
# Background

Vagrant is a tool for developer to provision reproducible development environments. We use vagrant to provide a developer sandbox for building the Script Language Container, which contains all necessary tools to accomplish this.

# Acceptance Criteria

- Add setup for libvirt
  - Resize image
  - DNS
  - Docker install

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #542 

### All Submissions:

* [x] Check if your pull request goes to the correct bash branch (develop or master)?
* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")

### When integrating to Develop branch:

1. [x] Remember to merge with "Squash and Merge"

### When integrating to Main branch:

1. [ ] Remember to merge with "Merge"

### CODE REVIEW

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# User Guide
+
+## Table of Content
+
+- [Using this Repositoriy](usage.md)
```
Comment (tomuben) [inline doc/user_guide/user_guide.md:5]:
```suggestion
- [Using this Repository](usage.md)
```

File: doc/user_guide/user_guide.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+- [Python Dataframe Support](py_dataframe.md)
+- [Using Vagrant as Development Environment](vagrant.md)
```
Comment (tomuben) [inline doc/user_guide/user_guide.md:7]:
```suggestion
- [Using Vagrant as development environment](vagrant.md)
```

File: vagrant/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Vagrant.configure("2") do |config|
+  config.vm.box = "generic/ubuntu2004"
+  config.vm.hostname = "script-languages"
+  config.vm.network :forwarded_port, guest: 8888, host: 8888
+  config.vm.network :forwarded_port, guest: 6666, host: 6666
+  config.vm.synced_folder ".", "/home/vagrant/vagrant", type: "rsync", rsync__exclude: ".git/"
+  config.vm.synced_folder "..", "/home/vagrant/script-languages-release", type: "rsync", rsync__exclude: [".git/",".build_output"], rsync__args: ["--verbose", "--archive", "-z", "--links", "--delete"], rsync__verbose: true
```
Comment (tomuben) [inline vagrant/Vagrantfile:7]: Does this mean the user does not need to call `vagrant rsync-auto`? Or does `vagrant rsync-auto` use this config lines to check what needs to be synced?

File: vagrant/Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    v.default_prefix = config.vm.hostname + "_"
+  end
+
+  config.trigger.before [:provision] do |trigger|
+    trigger.run = { inline: "vagrant rsync"}
+  end
+
+  #config.vm.provision "shell" do |s|
+  #   s.path="provisioning/resize-disk.sh"
+  #end
```
Comment (tomuben) [inline vagrant/Vagrantfile:24]: Was this only a test? Can it be removed? If not, add a comment please.

**Review by tomuben — COMMENTED** [review]
See my comments.

File: vagrant/Vagrantfile (hunk 3 above)
Comment (tkilias) [reply vagrant/Vagrantfile:7]: vagrant rsync and rsync-auto use these config lines to check what needs to be synced

File: vagrant/Vagrantfile (hunk 4 above)
Comment (tkilias) [reply vagrant/Vagrantfile:24]: ah, this actually needs uncommented

**Pull request merged**